### PR TITLE
Remove LSIF HTTP provider

### PR DESCRIPTION
**Pull request by efritz**
This is effectively dead code. We don't have any customers running an old instance of LSIF that we need to support. Customers that need LSIF support through the extensions must upgrade their instance.

### CODE REVIEW

File: shared/lsif/providers.ts [hunk 1]
```diff
@@ -30,55 +39,289 @@ export function createProviders(logger: Logger): Providers {
...
+>(
+    {
+        /** The current text document. */
+        doc,
+        /** The current hover position. */
+        position,
+        /** The GraphQL request query. */
+        query,
+        /** Additional query parameters. */
+        ...rest
```
Comment (creachadair) [inline shared/lsif/providers.ts]: I guess this is probably the existing behaviour, but should we consider doing this spread first so the trailing args can't override the essential keys?

**Review by creachadair — APPROVED** [review]
Just a couple minor things.

File: shared/lsif/providers.ts [hunk 2]
```diff
@@ -30,55 +39,289 @@ export function createProviders(logger: Logger): Providers {
...
+ * set of providers will use the GraphQL API.
+ *
+ * @param queryGraphQL The function used to query the GraphQL API.
+ */
+export function createGraphQLProviders(
+    queryGraphQL: QueryGraphQLFn<any> = sgQueryGraphQL
+): Providers {
+    return {
+        definition: asyncGeneratorFromPromise(definition(queryGraphQL)),
+        references: references(queryGraphQL),
```
[the remark this answers is not in the document]
Comment (efritz) [reply shared/lsif/providers.ts]: It's already a generator, the others return promises. Each return type makes the provider simpler (defs and hovers don't stream, refs do).

File: shared/lsif/providers.ts (hunk 1 above)
Comment (efritz) [reply shared/lsif/providers.ts]: So the spread operator MUST be the last for params, but not for object construction (which we do later in this method). It's a good point, updating.

File: shared/lsif/providers.ts [hunk 3]
```diff
@@ -30,55 +39,289 @@ export function createProviders(logger: Logger): Providers {
         return noopProviders
     }
 
-    const provider = selectProvider(logger)
+    logger.log('LSIF is enabled')
```
[the remark this answers is not in the document]
Comment (efritz) [reply shared/lsif/providers.ts]: Logging after providers are constructed (although they cannot currently throw as written, but that doesn't meant hey might not after a future change).

File: shared/lsif/providers.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (creachadair) [reply shared/lsif/providers.ts]: Ah, I see now. I tripped on the fact that we changed the top-level declaration from `async function* references` to `function references` but just boosted that inside. I'm guessing there was a reason for that, but it diminishes the legibility of intent IMO.

[the remark this answers is not in the document]
Comment (efritz) [reply shared/lsif/providers.ts]: It was so that those providers can close over the queryGraphQL function passed to the factory.

**Pull request merged**